controllers/applicationvolumereplication_controller.go

```diff
@@ -120,106 +126,263 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+	}
+
+	return &placementDecision, !requeue
+}
+
+func (r *ApplicationVolumeReplicationReconciler) isSubsriptionPausedForDR(labels map[string]string) bool {
+	return labels != nil &&
+		labels[RamenDRLabelName] != "" &&
+		strings.EqualFold(labels[RamenDRLabelName], "protected") &&
+		labels[subv1.LabelSubscriptionPause] != "" &&
+		strings.EqualFold(labels[subv1.LabelSubscriptionPause], "true")
+}
+
+// processPausedSubscription selects the target cluster from the subscription or
+// from the user selected cluster and restores all PVs (belonging to the subscription)
+// to the target cluster and then it unpauses the subscription
+func (r *ApplicationVolumeReplicationReconciler) processPausedSubscription(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) error {
+	r.Log.Info("Processing paused subscription", "name", subscription.Name)
+	// find target cluster (which can be the failover cluster)
+	homeClusterName := r.findNextHomeCluster(avr, subscription)
+
+	if homeClusterName == "" {
+		return errorswrapper.New("failed to find new home cluster")
+	}
+
+	err := r.deleteExistingManfiestWork(avr, subscription)
+	if err != nil {
+		return err
+	}
+
+	err = r.restoreBackedupPVs(subscription, homeClusterName)
+	if err != nil {
+		return err
+	}
+
+	return r.unpauseSubscription(subscription)
+}
+
+func (r *ApplicationVolumeReplicationReconciler) findVRGManifestWork(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) (bool, error) {
+	const notFound = false
+
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
+		mw := &ocmworkv1.ManifestWork{}
+		vrgMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, "vrg")
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: vrgMWName, Namespace: d.HomeCluster}, mw)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				r.Log.Info("ManifestWork not found for Subscription", "name", subscription.Name)
+
+				return notFound, nil
+			}
+
+			return notFound, errorswrapper.Wrap(err, "failed to retrieve manifestwork")
+		}
+
+		return found, nil
+	}
+
+	return notFound, nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) deleteExistingManfiestWork(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) error {
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
+		mw := &ocmworkv1.ManifestWork{}
+		vrgMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, "vrg")
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: vrgMWName, Namespace: d.HomeCluster}, mw)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+
+			r.Log.Error(err, "Could not fetch ManifestWork")
+
+			return errorswrapper.Wrap(err, "failed to retrieve manifestWork")
+		}
+
+		r.Log.Info("deleting ManifestWork", "name", mw.Name)
+
+		return r.Client.Delete(context.TODO(), mw)
```

Comment:
(future) We should also update VRG to "Secondary" and delete it later, so that we can heal the failed primary. The delete later part may need some status information from the MW back (which we do not have at present).

Reply:
When you say `"Update VRG to "Secondary"` you mean the failed primary?  If yes, then I am not sure if this is the path for it. Because it is possible that we are failing over because the primary is down and it may never comeback.  But to think about it, you might be right, because the MW for the VRG will be sitting in the hub waiting to be consumed by the failed primary. And at some point if it comes back, then it will find the MW waiting for it in the hub.  Alright, I'll create an issue for it.